CMakeLists.txt

```diff
@@ -13,7 +13,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #=============================================================================
-
+cmake_minimum_required(VERSION 3.17)
+project(tritonfilbackend LANGUAGES C CXX)
```

Comment:
`project` is declared at line 115. Can you help me understand why it's needed here? Also, if we could keep the project name consistent with what we've already done and ensure that the languages include CUDA iff necessary, that would be great.

Reply:
sorry, I missed that. It looks like it is not defined in the `TRITON_FIL_DOCKER_BUILD`. I can move the project definition to outside the if/else condition if you think that's appropriate. I ran into a build issue when trying to use `cmake --build` for server with Docker build:

- Build files have been written to: /tmp/tritonbuild/fil/build
+ cmake --build . -j80 -v -t install
Error: could not find CMAKE_PROJECT_NAME in Cache

Reply:
Ah! I understand, thank you. Yes, I think it would be good to move it into the if/else branch to avoid a double declaration of the project. If we could just update the name and (I think?) remove the C/CXX language specifications since we don't need to insist on C/CXX compiler availability when building with Docker, that should be perfect.

Thanks very much for the fix!

Reply:
Thanks for clarification. I made the change.